### DISCUSSION TITLE
Footnote support

### DISCUSSION THREAD

**Discussion by jermanuts** [withastro/starlight#2622] [Q&A] [answered]
### What version of `starlight` are you using?

29.2

### What is your idea?

Add footnotes support so readers can easily navigate. It is a great way to add supplemental or additional information to a specific word, phrase or sentence without interrupting the flow of a document.

### Why is this feature necessary?

Reduces the need for a lot of scrolls to find the relevant info mentioned at the end of the page.

### Do you have examples of this feature in other projects?

https://squidfunk.github.io/mkdocs-material/reference/footnotes

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment, marked as answer]
Thanks for your feedback :raised_hands:

Footnotes are supported out-of-the-box in Starlight using the same syntax in the provided link:

```md
Here is a footnote[^1] with some additional text after it.

[^1]: My reference.
```

Here is a preview of the result:

<img width="395" alt="image" src="https://github.com/user-attachments/assets/cfe17d3b-848b-4bbd-bd0a-3a478157c5fc">

Here is a live example on StackBlitz if needed: https://stackblitz.com/edit/github-dsfqdp?file=src%2Fcontent%2Fdocs%2Findex.mdx

**Reply by delucis** [reply]
I’ll add that this support is configurable via Astro’s [`markdown.remarkRehype`](https://docs.astro.build/en/reference/configuration-reference/#markdownremarkrehype) option, where you can pass [`remark-rehype` options](https://github.com/remarkjs/remark-rehype#options) to control footnote markup.

**Comment by jermanuts** [comment]
@HiDeoo Thanks for the swift reply, it wasn't documented in https://starlight.astro.build/ is there any reason?

**Reply by delucis** [reply]
I think probably because it’s a part of Astro’s defaults rather than an intentional choice and so it didn’t occur to us. The Markdown documentation in the Starlight docs are in general a bit in need of love — a lot of it dates from me just throwing lots of different markup on a page for style testing in the very early days of development 😁
